module/user-extension/helper.js

```diff
@@ -538,169 +573,307 @@ module.exports = class UserExtensionHelper {
-                    message : constants.apiResponses.USER_EXTENSION_NOT_FOUND,
-                    result : {}
+                    message: constants.apiResponses.USER_EXTENSION_FETCHED,
+                    result:
+                        _.merge(_.omit(userExtensionData[0], ["roles", "entityDocuments", "roleDocuments"]),
+                            { roles: _.isEmpty(roleMap) ? [] : Object.values(roleMap) })
                 });
+
+            } catch (error) {
+                return reject(error);
             }
+        })
+    }
 
-            let roleMap = {};
-            let entityMapToRelatedEntities = {};
-
-            if( userExtensionData[0].entityDocuments && userExtensionData[0].entityDocuments.length >0 ) {
-                
-                let projection = [
-                    "metaInformation.externalId", 
-                    "metaInformation.name", 
-                    "metaInformation.addressLine1",
-                    "metaInformation.addressLine2",
-                    "metaInformation.administration",
-                    "metaInformation.city",
-                    "metaInformation.country",
-                    "entityTypeId",
-                    "entityType"
-                ];
+    /**
+     * Get profile with entity details
+     * @method
+     * @name getEntities
+     * @param {Object} filterQueryObject - filtered data.
+     * @param {String} entityType - entity type
+     * @param {String} pageSize - Size of page.
+     * @param {String} pageNo - Recent page no.
+     * @param {String} search - search text.
+     * @returns {Object} 
+     */
 
-                for ( 
-                    let pointerToUser = 0; 
-                    pointerToUser < userExtensionData[0].entityDocuments.length ;
-                    pointerToUser ++ 
-                ) {
-                    let currentEntities = 
-                    userExtensionData[0].entityDocuments[pointerToUser];
-
-                    const relatedEntities = 
-                    await entitiesHelper.relatedEntities(
-                        currentEntities._id, 
-                        currentEntities.entityTypeId, 
-                        currentEntities.entityType, 
-                        projection
-                    );
+    static getEntities(filterQueryObject, entityType,pageSize,pageNo,searchText) {
+        return new Promise(async (resolve, reject) => {
+            try {
 
-                    if( relatedEntities.length > 0 ) {
-                        entityMapToRelatedEntities[currentEntities._id.toString()] = relatedEntities;
+                if( searchText !== "") {
+                    filterQueryObject["$or"] = [
+                        { "entityDocuments.metaInformation.name": new RegExp(searchText, 'i') },
+                    ];
+                }
+                if (entityType) {
+                    filterQueryObject['entityDocuments.entityType'] = entityType;
+                }
+                let queryObject = [
```

Comment:
@aman-tunerlabs - Where is the userId filter?

Reply:
@aks30  It is done in controllers only as filterQueryObject will have 
{
          userId : (req.params._id && req.params._id != "") ? req.params._id : req.userDetails.userId,
          status : constants.common.ACTIVE,
          isDeleted : false,
   }